docs/design/safety/terminology.md

```diff
@@ -0,0 +1,195 @@
+        -   [_Type safety_](#type-safety)
+        -   [_Initialization safety_](#initialization-safety)
+        -   [_Data-race safety_](#data-race-safety)
+    -   [_Memory safety bug_](#memory-safety-bug)
+    -   [_Memory-safe platform_ or _environment_](#memory-safe-platform-or-environment)
+    -   [_Memory-safe language_](#memory-safe-language)
+
+<!-- tocstop -->
+
+## Core Terminology
+
+### _Hazard_
+
+Unsafe coding construct that may lead to a bug or vulnerability.
+
+### _Bug_ or _defect_
+
+Reachable program behavior contrary to the author's intent.
+
+#### _Active bug_
+
+Buggy behavior that is actively occurring for users of the program.
+
+#### _Latent bug_
+
+Buggy behavior that does not currently occur for users, but is reachable.
+Behaviors that are reachable, and so _can_ happen, but don't happen today in
+practice _are always still bugs_!
+
+### _Safety_
+
+Absent a qualifier or narrow context, refers to _system safety_, and _safety
+engineering_. Always a property of a system or product as a whole, including
+human factors, etc.
+
+### _Code_, _software_, or _program safety_
+
+Invariants or limits on program behavior in the face of bugs.
+
+-   Very narrow and specific meaning.
+-   Often necessary but not sufficient for system safety.
+
+### _Safety bugs_
+
+Bugs where some aspect of program behavior has insufficient (often none)
+invariants or limits.
+
+For example, _undefined behavior_ definitionally has no invariant or limit, and
+reaching it is always a safety bug.
+
+### _Initial bug_
+
+The first behavior contrary to the author's intent, distinct from subsequent
+deviations.
+
+### _Fail-stop_
+
+The behavior of immediately terminating the program, minimizing any further
+business logic. This is in contrast to any form of "correct" program
+termination, continuing execution, or unwinding.
+
+## Vulnerability Terminology
+
+### _Vulnerability_ or _security vulnerability_
+
+A bug that creates the possibility for a malicious actor to subvert a program's
+intended behavior in a way that violates a security policy (for example,
+confidentiality, integrity, availability). Vulnerabilities are often exploitable
+manifestations of underlying bugs.
+
+### _Vulnerability defense_
+
+The set of strategies and techniques employed to reduce the risks posed by
+vulnerabilities arising from bugs. These strategies operate at different levels
+and have varying degrees of effectiveness.
+
+#### _Detecting_
+
+While still vulnerable, detecting or tracking the exploit of a bug. Requires
+_some_ invariant or limit, but very minimal.
+
+#### _Mitigating_
+
+Making a vulnerability significantly more expensive, difficult, or improbable to
+be exploited.
+
+#### _Preventing_ vulnerabilities
+
+While still a bug, making it impossible to be a vulnerability. Often this is
+done by defining behavior to [fail-stop](#fail-stop).
+
+#### _Ensuring_ correctness
+
+No longer a bug, much less a vulnerability.
```

Comment:
These two seem like incomplete definitions when read by themselves (which I expect these entries typically will be).